tests/platform_tests/test_advanced_reboot.py

```diff
@@ -42,6 +43,21 @@ def test_fast_reboot(request, get_advanced_reboot, verify_dut_health,
     advancedReboot.runRebootTestcase()
 
 
+@pytest.mark.usefixtures('get_advanced_reboot')
+def test_fast_reboot_from_other_vendor(duthosts,  rand_one_dut_hostname, request, get_advanced_reboot, verify_dut_health,
```

Comment:
I think it will be better to add this test in a new script. `test_fastboot_vendor_to_sonic.py`?

Reply:
The script test_advanced_reboot initiates various types of reboot, such as warm reboot, or fast reboot, with different parameters such as test_warm_reboot_mac_jump or test_cancelled_fast_reboot.
The new other vendor reboot test case is another parameter of advanced reboot and isn't different from other reboot types. nevertheless, all reboot types are using the infrastructure and fixtures of the advanced reboot script. Creating a new script will require duplication of fixtures and infrastructure.

Reply:
Unless I am missing something, creating a new test script will not require duplication of fixtures or infra. Any new test script will only have to import the fixtures, just how this script is doing.
None of the fixtures used by `tests/platform_tests/test_advanced_reboot.py` are actually defined in this file. Same will be the case for `test_fastboot_vendor_to_sonic.py`.

I am thinking that in future we will add more cases to non-SONiC to SONiC upgrade/reboot path. With tweaking parameters and also by doing a real (not just mock) upgrade. This deems a new category/script as it grows.